### PR TITLE
DMP-1044: Internal User Claim extract

### PR DESCRIPTION
**Pull request by hemantasharma1129**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [x] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/DMP-1044

### Change description ###
Added logic to extract Internal User Token claims.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/darts/authentication/controller/impl/AbstractUserController.java [hunk 1]
```diff
@@ -14,18 +14,19 @@
...
 import java.text.ParseException;
-import java.util.List;
 import java.util.Optional;
 
 @Component
 @RequiredArgsConstructor
 public abstract class AbstractUserController implements AuthenticationController {
 
-    private static final String EMAILS_CLAIM_NAME = "emails";
+    abstract Optional<String> parseEmailAddressFromAccessToken(String accessToken) throws ParseException;
```
Comment (karen-hedges) [inline src/main/java/uk/gov/hmcts/darts/authentication/controller/impl/AbstractUserController.java:23]: Should this method be moved to be below the variables?

File: src/main/java/uk/gov/hmcts/darts/authorisation/component/impl/UserIdentityImpl.java [hunk 2]
```diff
@@ -32,6 +38,8 @@ public String getEmailAddress() {
                 if (emailAddressObject instanceof String emailAddress && StringUtils.isNotBlank(emailAddress)) {
                     return emailAddress;
                 }
```
Comment (karen-hedges) [inline src/main/java/uk/gov/hmcts/darts/authorisation/component/impl/UserIdentityImpl.java:40]: Should there be an else statement to log an error unable to validate email address or something like that?

Comment (hemantasharma1129) [reply src/main/java/uk/gov/hmcts/darts/authorisation/component/impl/UserIdentityImpl.java:40]: Its already there right at the end of method.

File: src/main/java/uk/gov/hmcts/darts/authentication/controller/impl/AbstractUserController.java (hunk 1 above)
Comment (hemantasharma1129) [reply src/main/java/uk/gov/hmcts/darts/authentication/controller/impl/AbstractUserController.java:23]: Done

File: src/main/java/uk/gov/hmcts/darts/authorisation/component/impl/UserIdentityImpl.java (hunk 2 above)
Comment (karen-hedges) [reply src/main/java/uk/gov/hmcts/darts/authorisation/component/impl/UserIdentityImpl.java:40]: It throws an exception but does not log where the issue came from

**Review by karen-hedges — APPROVED** [review]

**Pull request merged**